pkg/cmd/extension/manager.go

```diff
@@ -116,36 +117,96 @@ func (m *Manager) list(includeMetadata bool) ([]extensions.Extension, error) {
 		if !strings.HasPrefix(f.Name(), "gh-") {
 			continue
 		}
-		var remoteUrl string
-		updateAvailable := false
-		isLocal := false
-		exePath := filepath.Join(dir, f.Name(), f.Name())
-		if f.IsDir() {
-			if includeMetadata {
-				remoteUrl = m.getRemoteUrl(f.Name())
-				updateAvailable = m.checkUpdateAvailable(f.Name())
-			}
-		} else {
-			isLocal = true
-			if !isSymlink(f.Mode()) {
-				// if this is a regular file, its contents is the local directory of the extension
-				p, err := readPathFromFile(filepath.Join(dir, f.Name()))
-				if err != nil {
-					return nil, err
-				}
-				exePath = filepath.Join(p, f.Name())
-			}
+		ext, err := m.parseExtensionDir(f, includeMetadata)
```

Comment:
What are your thoughts about separating the parsing of the extension directory from the retrieval of the metadata? Those two have been intertwined for awhile, but separating them will make it easier to speed up the list command since the retrieval of the metadata and the API requests are what take up so much time.

Reply:
definitely in favor of this but I think I need to personally punt given time constraints